clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -267,7 +268,8 @@ public class VenicePushJob implements AutoCloseable {
   public VenicePushJob(String jobId, Properties vanillaProps) {
     this.jobId = jobId;
     this.props = getVenicePropsFromVanillaProps(Objects.requireNonNull(vanillaProps, "VPJ props cannot be null"));
-    this.timeoutExecutor = Executors.newSingleThreadScheduledExecutor();
+    this.timeoutExecutor = Executors
+        .newSingleThreadScheduledExecutor(new DaemonThreadFactory(this.getClass().getName() + "-VPJTimeoutExecutor"));
```

Comment:
and call `timeoutExecutor.shutdownNow();` once the poll is completed, Lets not rely on the `close()` to stop this thread